### PR TITLE
Update to V2 contracts

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**
- Allows use for V1 and V2 governance contracts.
- Removed Rust nightly requirement.
- Updated dependencies.

Closes Issue #6

### CODE REVIEW

File: src/response/v2.rs [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+    }
+}
+
+impl EmissionBallotInfo {
+    fn email_text(&self) -> String {
+        format!(
+            "Creation Time: {}\n\
+            Voting Start Time: {}\n\
+            Voting End Time: {}\n\
+            Ammount: {}\n\
```
Comment (varasev) [inline src/response/v2.rs]: Please rename `Ammount` to `Amount`.

File: src/response/v2.rs [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+        };
+        let end_time = {
+            let uint = tokens[2].clone().to_uint().unwrap();
+            u256_to_datetime(uint)
+        };
+        let is_canceled = tokens[3].clone().to_bool().unwrap();
+        let is_finalized = tokens[4].clone().to_bool().unwrap();
+        let creator = tokens[5].clone().to_address().unwrap();
+        let memo = tokens[6].clone().to_string().unwrap();
+        let ammount = tokens[7].clone().to_uint().unwrap();
```
Comment (varasev) [inline src/response/v2.rs]: Could we convert the value in Wei to the value of POA coins? E.g., for `10441 POA` it displays `10441000000000000000000`. I think it's better to display `10441` instead. The amount also can be fractional.

File: README.md [hunk 3]
```diff
@@ -85,76 +85,115 @@ Then try to rebuild `poagov` using:
...
-        -h, --help        prints help information
-        -k                monitor the blockchain for ballots to change keys (same as --monitor=keys)
-            --latest      start monitoring for goverance events starting from the most recently mined block in the chain
-            --local       monitor voting contracts deployed to a locally running POA chain (same as using --network=local)
-        -p                monitor the change for ballots to change the proxy address (same as --monitor=proxy)
-            --push        send governance notifications via push notification
-            --sokol       monitor voting contracts deployed to the Sokol test network (same as using --network=sokol)
-        -t                monitor the chain for ballots to change the minimum threshold (same as --monitor=threshold)
-        -V, --version     prints version information
+        --core             monitor voting contracts deployed to the Core network
```
Comment (varasev) [inline README.md]: Please remove excess spaces from this line.

File: README.md [hunk 4]
```diff
@@ -85,76 +85,115 @@ Then try to rebuild `poagov` using:
...
-        --block-time <value>    the average time it takes to mine a new block
-        --monitor <value>       a comma-separated list of ballot types to monitor for governance events; the available values are: keys, threshold, proxy
-        --network <value>       the name of the network to monitor for ballots; the values available for this option are: core, sokol, local
-        --rpc <value>           the URL for the RPC endpoint
-        --start <value>         start monitoring for governance events at this block (inclusive)
-        --tail <value>          start monitoring for governance events for the `n` blocks prior to the last mined block in the chain
-
-# Setting up the Config File
+	--block-time <value>    the average number of seconds it takes to mine a new block
+        -n, --limit <value>     shutdown `poagov` after this many notifications have been generated
```
Comment (varasev) [inline README.md]: This line also contains excess spaces. Seems copy-pasted from terminal :)

File: README.md [hunk 5]
```diff
@@ -85,76 +85,115 @@ Then try to rebuild `poagov` using:
...
+email text will be logged regardless of whether or not the `--email` flag is
+set.
+
+# Setting up the `.env` File
 
 When the `poagov` CLI tool is run, the process' environment variables are
-loaded via a `.env` file. An example `.env` file can be found at
-`sample.env`. Before running this tool change the name of `sample.env` to
-`.env` using:
+loaded via an `.env` file. Before running `poagove` copy `sample.env` into
```
Comment (varasev) [inline README.md]: Typo in `poagov` (poagove).

**Comment by varasev** [comment]
@DrPeterVanNostrand Thank you for the new version. I tried to build this from source and to launch - it seems to be working fine. There are some minor typos I mentioned above - please fix them before merging.

**Comment by varasev** [comment]
@DrPeterVanNostrand Please take a look at the comments above.

**Comment by DrPeterVanNostrand** [comment]
Thanks for reviewing. I made the changes that you asked for over the weekend. I'll push the updates and create a new release tomorrow.

**Pull request merged**